dart/lib/src/scope.dart

```diff
@@ -136,4 +138,28 @@ class Scope {
   void removeExtra(String key) {
     _extra.remove(key);
   }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is Scope &&
+          runtimeType == other.runtimeType &&
+          _level == other.level &&
+          _transaction == other.transaction &&
+          _user == other.user &&
+          IterableEquality().equals(_fingerprint, other.fingerprint) &&
+          IterableEquality().equals(_breadcrumbs, other.breadcrumbs) &&
+          MapEquality().equals(_tags, other.tags) &&
+          MapEquality().equals(_extra, other.extra);
+
+  @override
+  int get hashCode =>
+      _level.hashCode ^
+      _transaction.hashCode ^
+      _user.hashCode ^
+      _fingerprint.hashCode ^
+      _breadcrumbs.hashCode ^
+      _tags.hashCode ^
+      _extra.hashCode ^
+      _options.hashCode;
```

Comment:
mm why do we need this?

Reply:
to test scopes equality in tests.

Reply:
Could we move this code to the tests instead? I'd avoid having this here if we don't need to maintain as production code (like putting scopes in as Map keys)

Reply:
I'm not sure I can without changing this kind of tests : 
```dart
test('should capture event', () {
      hub.captureEvent(fakeEvent);
      verify(
        client.captureEvent(fakeEvent,
            scope: Scope(options), stackFrameFilter: null),
      ).called(1);
    });
```
If we don't override the `==`operator the test will fail as it will only compare scope reference. 
To remove the need of this override, I could change the test to : 
```dart
test('should capture event', () {
      hub.captureEvent(fakeEvent);
      verify(
        client.captureEvent(fakeEvent,
            scope: anyNamed('scope'), stackFrameFilter: null),
      ).called(1);
    });
```
but it less precise. WDYT?

Reply:
Based on the name of the tests, we're not really asserting anything about the scope, is that right? We just test that an event was captured?

For a test that requires validating equality of the argument, is there a arg validator that takes a callback?
like:

`arg.is(s => s.a == expected.a)` ?

Reply:
this works :
```dart
bool scopeEquals(Scope a, Scope b) {
      return identical(a, b) ||
        a is Scope && b is Scope &&
          a.runtimeType == b.runtimeType &&
          a.level == b.level &&
          a.transaction == b.transaction &&
          a.user == b.user &&
          IterableEquality().equals(a.fingerprint, b.fingerprint) &&
          IterableEquality().equals(a.breadcrumbs, b.breadcrumbs) &&
          MapEquality().equals(a.tags, b.tags) &&
          MapEquality().equals(a.extra, b.extra);
    }

    test('should capture event with correct scope', () {
      hub.captureEvent(fakeEvent);
      expect(
          scopeEquals(
              verify(
                client.captureEvent(
                  fakeEvent,
                  scope: captureAnyNamed('scope') /*  Scope(options)*/,
                  stackFrameFilter: null,
                ),
              ).captured.first,
              Scope(options)),
          true);
    });
```
but we still depend on package collection

Reply:
Add the dependency as a test dependency and we're golden

Reply:
that equals funciton could be an extension method on the scope class, defined on the test package?

Reply:
the "dartiest" way ! 😄

Reply:
but needs Dart >=2.6 :(

Reply:
I'd also prefer to keep this type of code needed only for tests, on the test packages

Reply:
✅  moved in the test side, as a simple method for now
We should decide if we update the SDK constraints to use extensions and other "modern dart" features :)

Reply:
https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#260---2019-11-05
Completing 1 year:

https://github.com/getsentry/sentry-dart/issues/115